### PR TITLE
Add tests specifically for checking student access

### PR DESCRIPTION
**Pull request by carsongee**
This just adds several tests similar to testing for staff access, but for our student roles.

### CODE REVIEW

**Comment by carsongee** [comment]
@amir-qayyum-khan do you mind reviewing this for me please?

**Comment by amir-qayyum-khan** [comment]
yes sure @carsongee

File: pylti/tests/test_flask.py [hunk 1]
```diff
@@ -267,6 +267,46 @@ def test_access_to_oauth_resource_staff_only_as_unknown_role(self):
         self.app.get(new_url)
```
Comment (amir-qayyum-khan) [inline pylti/tests/test_flask.py]: @carsongee  please make it admin_url or administrator_url

File: pylti/tests/test_flask_app.py [hunk 2]
```diff
@@ -142,6 +142,19 @@ def initial_staff_route(lti):
     return "hi"
 
 
+@app.route("/initial_student", methods=['GET', 'POST'])
+@lti_flask(error=error, request='initial', role='student', app=app)
+def initial_student_route(lti):
+    # pylint: disable=unused-argument,
+    """
+    access route with 'initial' request and 'student' role
```
Comment (amir-qayyum-khan) [inline pylti/tests/test_flask_app.py]: Please add period (full stop) after each doc string.

https://www.python.org/dev/peps/pep-0257/#one-line-docstrings

File: pylti/tests/test_flask.py [hunk 3]
```diff
@@ -267,6 +267,46 @@ def test_access_to_oauth_resource_staff_only_as_unknown_role(self):
...
+        consumers = self.consumers
+        url = 'http://localhost/initial_unknown?'
+        staff_url = self.generate_launch_request(
+            consumers, url, roles='Staff'
+        )
+        self.app.get(staff_url)
+        self.assertTrue(self.has_exception())
+
+    def test_access_to_oauth_resource_student_as_unknown(self):
+        """Verify staff doesn't have access to student only"""
```
Comment (amir-qayyum-khan) [inline pylti/tests/test_flask.py]: Please add period (full stop) after each doc string.

https://www.python.org/dev/peps/pep-0257/#one-line-docstrings

File: pylti/tests/test_flask.py [hunk 4]
```diff
@@ -267,6 +267,46 @@ def test_access_to_oauth_resource_staff_only_as_unknown_role(self):
...
+        self.assertFalse(self.has_exception())
+
+        student_url = self.generate_launch_request(
+            consumers, url, roles='Student'
+        )
+        self.app.get(student_url)
+        self.assertFalse(self.has_exception())
+
+    def test_access_to_oauth_resource_student_as_staff(self):
+        """Verify staff doesn't have access to student only"""
```
Comment (amir-qayyum-khan) [inline pylti/tests/test_flask.py]: Please add period (full stop) after each doc string.

https://www.python.org/dev/peps/pep-0257/#one-line-docstrings

File: pylti/tests/test_flask.py [hunk 5]
```diff
@@ -267,6 +267,46 @@ def test_access_to_oauth_resource_staff_only_as_unknown_role(self):
         self.app.get(new_url)
         self.assertTrue(self.has_exception())
 
+    def test_access_to_oauth_resource_student_as_student(self):
+        """
+        Verify that the various roles we consider as students are students
```
Comment (amir-qayyum-khan) [inline pylti/tests/test_flask.py]: Please add period (full stop) after each doc string.

https://www.python.org/dev/peps/pep-0257/#one-line-docstrings

**Comment by amir-qayyum-khan** [comment]
@carsongee  other then few changes i mention it looks good.
Thanks

**Comment by carsongee** [comment]
I fixed up all those instances, and found several more doc strings that weren't quite right.

**Comment by amir-qayyum-khan** [comment]
Looks great :+1:

**Pull request merged**